evaluator/functions.go

```diff
@@ -1161,6 +1165,27 @@ func waitFn(tok token.Token, args ...object.Object) object.Object {
 	return cmd
 }
 
+// kill(`sleep 10 &`)
+func killFn(tok token.Token, args ...object.Object) object.Object {
+	err := validateArgs(tok, "kill", args, 1, [][]string{{object.STRING_OBJ}})
+	if err != nil {
+		return err
+	}
+
+	cmd := args[0].(*object.String)
+
+	if cmd.Cmd == nil {
+		return cmd
+	}
+
+	errCmdKill := cmd.Kill()
+
+	if errCmdKill != nil {
+		return newError(tok, "Failed to kill command %s ", cmd.Inspect())
```

Comment:
@mingwho can we also add the actual error message (`err.CmdKill.Error`) here?

Reply:
👍 fixed